surveys/2022-wg-debugging-survey/questions.md

```diff
@@ -0,0 +1,361 @@
+> it would be interesting to know that.
+
+### Which programming language(s) do you regularly use a debugger with?
+Type: select all that apply
+
+- C
+- C++
+- Java
+- JavaScript
+- Python
+- Go
+- C#
+- Other
+
+> *Justification:*
+>
+> Debugging experiences vary widely by language - debugging a interpreted
+> language like Python is very different from a compiled language like C/C++ -
+> this will inform user expectations with Rust debuggers.
+
+### How does debugging in language X compare to Rust?
+Type: freeform text
+
+> *Justification:*
+>
+> What can Rust's debugging experience learn from other languages.
+
+### Do you often use a debugger with Rust?
+Type: select one
+
+- Daily or nearly so
+- Weekly or nearly so
+- Monthly or nearly so
+- Rarely
+
+> *Justification:*
+>
+> Knowing how often a debugger is used with Rust is valuable, but need to know
+> whether that is because of the quality of the experience or because it
+> is/isn't often needed.
+
+### How do you start a debugger (when debugging Rust)?
+Type: select all that apply
+
+- Core dumps
+- Attaching to a process
+- Launching executable
+- From an IDE
+
+> *Justification:*
+>
+> User experience for starting a debugger in either of these ways could be
+> improved if they were particularly common or uncommon.
+
+### What debuggers have you heard of?
+Type: select all that apply
+
+- `gdb`
+- `lldb`
+- Visual Studio
+- WinDBG
+- `rr`
+
+> *Justification:*
+>
+> If users aren't aware of certain types of debugger than raising awareness or
+> making it easier to start these debuggers might be valuable.
+
+### What debuggers do you use with Rust?
+Type: select all that apply
+
+- `gdb`
+- `lldb`
+- Visual Studio
+- WinDbg
+- `rr`
+
+> *Justification:*
+>
+> Debuggers that are actually used is useful.
+
+### Which debugger interfaces do you use with Rust?
+Type: select all that apply
+
+- Editor extension (e.g. `termdebug`)
+- Built-in IDE debugger
+- Terminal line interface (e.g. `gdb`)
+- Terminal UI (e.g. `gdb`'s TUI)
+- Web interface (e.g. Pernosco)
+- Separate debugging program
+
+> *Justification:*
+>
+> Interface used for a debugger materially changes the experience of using a
+> debugger and the advantages/disadvantages of it.
+
+### What pain points have you experienced using a debugger with Rust?
+Type: select all that apply
+
+- Often step too many times and need to restart from scratch
+- Don't have a source view of the current line being run
+- Unintuitive interface/errors
+- Expression parser doesn't support expressions that you want to write
+- Optimized-out code
+- Conditional breakpoints don't trigger
+- Difficult to trace dataflow
+- Too much detail/irrelevant information (i.e. assembly views)
+- Incorrect information displayed (i.e. line numbers, variable values)
+- Current line number jumps around unexpectedly
+- Values aren't visualized as expected (e.g. `Vec<T>` is shown as a pointer,
```

Comment:
I would phrase this as something like "sub-optimal representations of values" because "as expected" is a bit ambiguous and "visualisation" is debugger jargon

Reply:
Rephrased this